### PR TITLE
Trigger WebSocket error event

### LINKED ISSUE #7861: Don't ignore errors when sending data over websockets
In `WebSocket::send` we ignore the result of actually sending the data over the network. That causes us to fail tests, funnily enough :) Following the [spec](https://tools.ietf.org/html/rfc6455#section-10.7) for dealing with invalid data, we should be dispatching a `CloseTask` to ensure that the connection is terminated, and additionally for certain cases of errors we should be implementing the `fail the websocket connection` algorithm which causes the `close the websocket connection` to dispatch an `error` event at the `WebSocket` object. One example is when there are [errors processing UTF-8 data](https://tools.ietf.org/html/rfc6455#section-8.1), which corresponds to a case in the [error result](http://doc.servo.org/websocket/result/enum.WebSocketError.html) from calling send!

Test: `./mach test-wpt tests/wpt/web-platform-tests/websockets/interfaces/WebSocket/events/018.html`
Code: components/script/dom/websocket.rs

### PR DESCRIPTION
**Pull request by jdm**
Trigger a WebSocket error after receiving an invalid message from the server. Rebased from #8868. Fixes #7861.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9757)

### CODE REVIEW

**Comment by KiChjang** [comment]
```
Ran 4606 tests finished in 229.0 seconds.
  • 4602 ran as expected. 1374 tests skipped.
  • 3 tests unexpectedly okay
  • 4 tests had unexpected subtest results

Tests with unexpected results:
  ▶ OK [expected TIMEOUT] /websockets/constructor/010.html

  ▶ Unexpected subtest result in /websockets/constructor/010.html:
  │ FAIL [expected TIMEOUT] WebSockets: protocol in response but no requested protocol
  │   → assert_unreached: error Reached unreachable code
  │ 
  │ ws.onclose<@http://web-platform.test:8000/websockets/constructor/010.html:3:69
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1405:1

  ▶ OK [expected TIMEOUT] /websockets/constructor/011.html

  ▶ Unexpected subtest result in /websockets/constructor/011.html:
  └ PASS [expected TIMEOUT] WebSockets: protocol mismatch

  ▶ Unexpected subtest result in /websockets/cookies/005.html:
  │ FAIL [expected PASS] WebSockets: setting HttpOnly cookies in ws response, checking ws request
  │   → assert_unreached: error Reached unreachable code
  │ 
  │ t</ws.onclose<@http://web-platform.test:8000/websockets/cookies/005.html:21:54
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1405:1

  ▶ OK [expected TIMEOUT] /websockets/opening-handshake/005.html

  ▶ Unexpected subtest result in /websockets/opening-handshake/005.html:
  │ FAIL [expected TIMEOUT] WebSockets: proper first line
  │   → assert_equals: expected true but got false
  │ 
  │ ws.onopen</ws.onclose<@http://web-platform.test:8000/websockets/opening-handshake/005.html:6:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1405:1
```

Looks like test expectation updates are in order.

**Comment by jdm** [comment]
@bors-servo: retry
* Unclear. All tests passed, but the actual command returned a non-zero code.

File: tests/wpt/metadata/websockets/opening-handshake/005.html.ini [hunk 1]
```diff
@@ -1,6 +1,5 @@
 [005.html]
   type: testharness
-  expected: TIMEOUT
   [WebSockets: proper first line]
-    expected: TIMEOUT
+    expected: FALSE
```
Comment (KiChjang) [inline tests/wpt/metadata/websockets/opening-handshake/005.html.ini]: Perhaps this expected: FALSE is causing an issue?

**Comment by Ms2ger** [comment]
@bors-servo r=jdm

I fixed an `expected: FALSE` in `tests/wpt/metadata/websockets/opening-handshake/005.html.ini`.

**Pull request merged**